### PR TITLE
Admin panel

### PR DESCRIPTION
**Pull request by elakorkmaz**

### CODE REVIEW

File: frontend/scss/application.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@import "bootstrap";
+@import "vendor";
+@import "components";
+
+.navbar-brand {
```
Comment (izelnakri) [inline frontend/scss/application.scss]: these should go to bootstrap override I think, probably the bootstrap/_type.scss file.

File: frontend/scss/application.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  font: 'Segoe UI';
+  font-size: 58;
+  font-weight: bold;
+  color: black
+}
+
+.nav.navbar-nav.float-xs-right {
```
Comment (izelnakri) [inline frontend/scss/application.scss]: no custom styles, move it to perhaps an fd-application-header component

File: frontend/scss/application.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  margin-top: 2.5%;
+}
+
+.container {
+}
+
+.navbar.navbar-fixed-bottom.navbar-lg {
```
Comment (izelnakri) [inline frontend/scss/application.scss]: same here, try to do this with bootstrap first

File: index.js [hunk 4]
```diff
@@ -36,38 +31,64 @@ app.use(methodOverride((req, res) => {
...
-    res.render('/menus/show', { meal: meal });
-  });
-});
+// menu pages ------------------------------------------------------------------
 
 app.get('/menus/:slug', (req, res) => {
   db.Menu.findOne({
     where: {
-      slug: req.params.slug
+      slug: req.params.slug,
```
Comment (izelnakri) [inline index.js]: remove last comma

File: index.js [hunk 5]
```diff
@@ -36,38 +31,64 @@ app.use(methodOverride((req, res) => {
...
   }).then((menu) => {
-    res.render('menus/show', { menu: menu });
-  }).catch((error) => {
-    res.status(404).end();
+    db.UserMenu.findAndCountAll({
+      where: {
+        MenuId: menu.id
+      }
+    }).then((result) => {
+      res.render('menus/show', { menu: menu, result: result, user: req.session.user });
```
Comment (izelnakri) [inline index.js]: hmm result not a very good name

File: index.js [hunk 6]
```diff
@@ -36,38 +31,64 @@ app.use(methodOverride((req, res) => {
...
+    });
   });
 });
 
+// posting a vote --------------------------------------------------------------
+
+app.post('/menus/:id/votes', (req, res) => {
+  db.Menu.findById(req.params.id).then((menu) => {
```
Comment (izelnakri) [inline index.js]: you are not using this query block, please remove it

File: migrations/20161205103148-remove-column-number.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.createTable('Votes', {
```
Comment (izelnakri) [inline migrations/20161205103148-remove-column-number.js]: you create Votes table twice?

File: migrations/20161212102532-add-column-admin.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
+'use strict';
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    return queryInterface.createTable('Users', {
```
Comment (izelnakri) [inline migrations/20161212102532-add-column-admin.js]: two migrations that create the Users table?

File: routes/user.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+// register a user ---------------------------------------------------------- //
+
+router.get('/register', (req, res) => {
+  if (req.session.user) {
+    res.redirect('/');
+  }
+  res.render('users/new');
+});
+
+router.post('/register', (req, res) => {
```
Comment (izelnakri) [inline routes/user.js:15]: I would move /register, login routes to authentication route but its ok

File: views/admin/new.pug [hunk 10]
```diff
@@ -1,16 +1,18 @@
...
-  .form-group
-    label Menu content:
-    textarea.form-control(rows='20', name='content')
-  .form-group.text-xs-right
-    button.btn.btn-success.btn-lg(type='submit') Create Menu
+  .container
+    .row
+      .col-xs-6.offset-xs-3
+        h2 Register an administrator
+        form(action="/admin/register" method="POST")
```
Comment (izelnakri) [inline views/admin/new.pug]: you dont need an admin register!!!!

File: views/menus/edit.pug [hunk 11]
```diff
@@ -0,0 +1,23 @@
+extends ../application.pug
+
+block content
+
+  .container
+      .row
+        .col-xs-8.offset-xs-2
+          h2= "Edit " + menu.title
+          form(action="/admin/menus/" + menu.id method="POST")
```
Comment (izelnakri) [inline views/menus/edit.pug:9]: this should be a PUT request not POST

File: index.js (hunk 6 above)
Comment (elakorkmaz) [reply index.js]: I do use it

File: views/menus/edit.pug (hunk 11 above)
Comment (elakorkmaz) [reply views/menus/edit.pug:9]: but then it doesn't work

Comment (izelnakri) [reply views/menus/edit.pug:9]: you are not doing it right then, check methodOverride middleware

Comment (izelnakri) [reply views/menus/edit.pug:9]: there should be the POST but also hidden PUT field

Comment (elakorkmaz) [reply views/menus/edit.pug:9]: now it works with post and hidden input

**Pull request merged**